lib/parser.js

```diff
@@ -900,7 +900,22 @@ module.exports = function parse (line) {
 
     // replace custom keywords
     Object.keys(replacements).forEach(function(key) {
-        line = line.replace(new RegExp(key,'g'), replacements[key])
+
+        // preprocess tokens and replace special names
+        line = line.replace(/'[^']+'|\S+/g, function (match, part, str) {
```

Comment:
Since the same regex is used below for splitting keys, why not just modify keys below?

Reply:
The way we return “line” in certain parts, particularly when things aren’t dogescript it ends up messing with the leading whitespaces. I tried it just modifying the keys and ended up finding that the return of line messed things up. 

If we don’t care about preserving white space then we can just use joinKeys instead of line everywhere and simplify stuff! 

I actually prefer that option , and since not many consumers will use 2.4 I guess those passthrough returns wouldn’t matter. 

Another option would be to assume everything is dogescript and have a ‘nodoge’ mode that doesn’t modify the text.

Reply:
Whitespace isn't handled all that well right now, but I think it would be better to have similar output to older versions for now

Reply:
Ok! We can leave the fiddling with line then! If you can think of a cleaner regex we can try that too.